src/utils/streams.js

```diff
@@ -0,0 +1,218 @@
+    static inputOutput(filePath) {
+        File.getReadStream(filePath)
+            .pipe(through2((chunk, enc, callback) => {
+                output(chunk);
+                callback();
+            }));
+    }
+
+    static transformFile(filePath, isSaveToFile) {
+        let csvColumnsName;
+        let outputBuffer = {
+            maxLength: 1,
+            buffer: []
+        };
+        let isFirstChunk = true;
+        let isSupported = supportedFileFormats.some(fileFormat => File.checkExtname(filePath, fileFormat));
+
+        if (!isSupported) {
+            output(config.messages.notSupportedFileFormats, supportedFileFormats.join(', '));
+            return;
+        }
+
+        let readStream = File.getReadStream(filePath);
+        let writeStream = (isSaveToFile) ? (File.getWriteStream(File.getPathForSaveJSON(filePath))) : (process.stdout);
+
+        readStream
+            .pipe(split2())
+            .pipe(through2((chunk, enc, callback) => {
+                if (!csvColumnsName) {
+                    csvColumnsName = ImporterUtils.getCSVColumnsName(chunk.toString());
+                    callback(null, '[');
+                } else {
+                    try {
+                        callback(null, ImporterUtils.convertCSVRowToJSON(chunk.toString(), csvColumnsName));
+                    } catch (err) {
+                        callback(err);
+                    }
+                }
+            }))
+            .pipe(through2(function bufferProcessing(chunk, enc, callback) {
+                outputBuffer.buffer.push(chunk);
+                if (outputBuffer.buffer.length > outputBuffer.maxLength) {
+                    this.push(outputBuffer.buffer.shift());
+                    if (!isFirstChunk) {
+                        this.push(',');
+                    }
+                    isFirstChunk = false;
+                }
+                callback();
+            }))
+            .pipe(writeStream);
+
+        readStream.on('end', () => {
+            writeStream.write(outputBuffer.buffer.join(','));
+            writeStream.write(']');
+        });
+    }
+
+    static transform() {
+        output(config.messages.typeDataToTransform);
+
+        process.stdin
+            .pipe(Streams.transformToUpperCase())
+            .pipe(through2((chunk, enc, callback) => {
+                process.stdin.end();
+                output(chunk);
+                callback();
+            }));
+    }
+
+    static cssBundler(dirPath) {
+        const bundleName = 'bundle.css';
+        let emitter = new EventEmitter();
+        let bundleFilePath = path.join(dirPath, bundleName);
+        let writeStream = File.getWriteStream(bundleFilePath);
+        let filesQueue = new Queue(Streams.cssBundlerFileHandler.bind(null, writeStream), 1);
+        let isAddFilesEnded = false;
+
+        File.getFilesFromPath(dirPath, '.css', [bundleName], emitter);
+
+        filesQueue.onQueueEmpty(() => {
+            if (isAddFilesEnded) {
```

Comment:
Maybe endless queue is not a good idea...

Reply:
It's not an endless queue. Handler inside `onQueueEmpty()` will be called everytime
when `Queue` will be empty, but flag `isAddFilesEnded` will be `true` only when method `File.getFilesFromPath()` found all files in a `dirPath`.

Reply:
A quite complex logic for a pretty simple case, don't you think so?

Reply:
Agree. Removed at all